pkg/providers/oidc/provider.go

```diff
@@ -96,3 +100,8 @@ func (p *discoveryProvider) PKCE() PKCE {
 		CodeChallengeAlgs: p.codeChallengeAlgs,
 	}
 }
+
+// SupportedSigningAlgs returns the discovered provider signing algorithms.
+func (p *discoveryProvider) SupportedSigningAlgs() []string {
```

Comment:
Just a question, why do we not make the fields in the `discoveryProvider` struct exported if we are just doing to do it via a method anyways

Reply:
The discovery provider is an interface, in other areas of the code we use a fake/mock discovery provider rather than the real one to simulate the values we want. It's easier typically to use an interface in these scenarios rather than populating some existing struct. IMO anyway